cmd/snapctl/main.go

```diff
@@ -57,27 +57,48 @@ func init() {
 	url := flURL.Value
 	ver := flAPIVer.Value
 	secure := false
+	var arg []string
 
 	for idx, a := range os.Args {
 		switch a {
 		case "--url":
-			if err := f1.Parse(os.Args[idx : idx+2]); err == nil {
+			if len(os.Args) >= idx+2 {
+				arg = os.Args[idx : idx+2]
+			} else {
+				break
+			}
+			if err := f1.Parse(arg); err == nil {
 				url = *prtURL
 			}
 		case "-u":
-			if err := f1.Parse(os.Args[idx : idx+2]); err == nil {
+			if len(os.Args) >= idx+2 {
+				arg = os.Args[idx : idx+2]
+			} else {
+				break
+			}
+			if err := f1.Parse(arg); err == nil {
 				url = *prtU
 			}
 		case "--api-version":
-			if err := f1.Parse(os.Args[idx : idx+2]); err == nil {
+			if len(os.Args) >= idx+2 {
```

Comment:
You can't check before the switch without moving the "--insecure" option outside of the switch since it's a boolean flag and only requires that idx is in range not idx+2. Checking outside would result in skipping --insecure if it was the last flag given.